src/collectors/redisstat/redisstat.py

```diff
@@ -268,9 +261,42 @@ def _get_info(self, host, port, auth):
         # Pull full stats. Use execute_command for backwards compatibility
         # with older redis clients.
         info = client.execute_command('INFO', 'all')
-        del client
         return info
 
+    def _reset_stats(self, host, port, auth):
+        """Resets statistics on the specified Redis instance
+
+:param str host: redis host
+:param int port: redis port
+:rtype: dict
+
+        """
+        client = self._client(host, port, auth)
+        if client is None:
+            return None
+
+        # Reset statistics. Use execute_command for backwards compatibility
+        # with older redis clients.
+        client.execute_command('CONFIG RESETSTAT')
```

Comment:
I think the generic logging is enough.

In the longer term, we probably should provide a generic hook in the main collection loop for logging these failures in a generic way.